### PR TITLE
Incident Monitor: production-mirroring adversarial scenario packs (TAN-487)

### PR DESCRIPTION
**Pull request by tacshade**
## What changed?

Implements **TAN-487** — versioned, data-driven adversarial scenario packs that run in dry-run/sandbox mode against the Incident Monitor destination router's route preview, approval gate, and readiness gates. They surface authority-boundary, escalation, and safety gaps before production and **never mutate external systems**.

- **Data model** (`tandem-incident-monitor::scenarios`): `IncidentMonitorScenarioPack` / `Scenario` / `Input` / `Expectation` plus a built-in **versioned default pack** (embedded JSON) covering the eight adversarial categories: regulatory escalation, cross-system dispute, prompt injection / instruction conflict, excessive agency / tool overreach, low-confidence / missing-evidence, cross-tenant / missing-tenant, unsafe / unready destination, and recurring denied-action.
- **Dry-run runner** (`incident_monitor_scenarios`): synthesizes an *untrusted* route context per scenario (so forged tenant/project bindings are stripped exactly as a real forged report would be), computes a route preview, and reports `pass` / `fail` / `blocked` with per-assertion detail, evidence refs, and a `finding_id` on failure. A passing scenario means the operator's config produced the expected control behavior; a failing one surfaces a governance gap; an unroutable case is reported `blocked` (not evaluated).
- **Endpoints**: `GET`/`POST /incident-monitor/security/scenario-packs` — admin-only (full API token; scoped intake keys rejected). `POST` accepts an optional custom `pack` and `scenario_ids` filter. Results are also folded into the Phase-8 assessment report under `sections.adversarial_scenario_packs` and persisted in its evidence pack.
- **Docs**: `docs/incident-monitor-adversarial-scenario-packs.md` — states packs run only in dry-run/sandbox and must be authorized and bounded to systems you operate.

## Why?

Phase-7 controlled probes verify individual Tandem controls; the operating model also needs production-mirroring adversarial scenarios that exercise the full routing/approval/readiness path against realistic edge cases, with governed, exportable evidence — the Phase-8 goal in TAN-487.

## How to test

- [ ] `pnpm exec tsc --noEmit` — n/a (no TS changed)
- [x] (Rust) `cargo test -p tandem-incident-monitor --lib scenarios` and `cargo test -p tandem-server --lib -- incident_monitor` — 181 server incident-monitor tests + crate tests pass, including new tests for pack parsing, dry-run execution, approval-required, unsafe-destination fail-closed, prompt-injection, scenario filtering, admin-auth, gap-surfacing, and report evidence persistence
- [x] `cargo fmt --all --check`, `scripts/ci-file-size-check.sh`, `scripts/check-incident-monitor-terminology.mjs` all clean

## Security considerations

- [x] No secrets added
- [x] Permissions / filesystem rules reviewed — scenario packs are an admin-only, read-only surface; the runner only reads config and computes a route preview (`mutates_external_systems` is always false); a scoped intake key is rejected with 403 and, when an admin token is configured, missing/invalid tokens are rejected with 401

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01VpMaKGfZ4zqyyYFMJ9eFJH)_

### CODE REVIEW

File: crates/tandem-server/src/incident_monitor_scenarios.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        all_ok &= ok;
+        assertions.push(assertion(
+            "blocked",
+            json!(expected_blocked),
+            json!(actual),
+            ok,
+        ));
+    }
+    if let Some(expected_approval) = expect.approval_required {
+        if preview.effective_destination_ids.is_empty() {
```
[the remark this answers is not in the document]
Comment (tacshade) [reply crates/tandem-server/src/incident_monitor_scenarios.rs]: Fixed in 76ed7c6. The approval assertion is now treated as not-evaluable (reported `blocked`) whenever the preview is blocked or has no routable destination — not only when `effective_destination_ids` is empty. So a configured-but-unready destination (`blocked: true` with non-empty destinations) no longer yields a false `pass` on `approval_required` alone. Added a regression test using an unready GitHub destination.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/tandem-server/src/incident_monitor_scenarios.rs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    })
+}
+
+fn assertion(name: &str, expected: Value, actual: Value, ok: bool) -> Value {
+    json!({ "name": name, "expected": expected, "actual": actual, "ok": ok })
+}
+
+fn scenario_submission(scenario: &IncidentMonitorScenario) -> IncidentMonitorSubmission {
+    let input = &scenario.input;
+    IncidentMonitorSubmission {
```
[the remark this answers is not in the document]
Comment (tacshade) [reply crates/tandem-server/src/incident_monitor_scenarios.rs]: Fixed in 76ed7c6. Root-caused a bit deeper: the runner built the context from a synthetic *untrusted* report, and the router's enrichment strips `source_kind` (and tenant/project) for untrusted reports — so even forwarding it wouldn't have reached route matching. Reworked the runner to set the routing signals directly on the context (mirroring the assessment report's own route-preview helper), so `source_kind` / tenant / risk all flow into route matching. Added a regression test asserting `source_kind: "ci"` routes to a `ci`-specific route's destination.

---
_Generated by [Claude Code](https://claude.ai/code)_
